internal/currency/fetch.go

```diff
@@ -1,42 +1,80 @@
 package currency
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
+	"net/http"
+	"net/url"
 	"time"
-
-	currconv "github.com/kitloong/go-currency-converter-api/v2"
 )
 
+// Config is our config file storing out API URL, and our API Key.
+type Config struct {
+	CCKey string
+	URL   string
+}
+
+// Type ConverterService has all of the handler functions as reciever's
 type ConverterService struct {
 	APIKey string
+	URL    string
 }
 
-func NewService(APIKey string) (ConverterService, error) {
-	if APIKey == "" {
-		return ConverterService{}, errors.New("APIKey cannot be empty")
+// New service takes a config, and returns is the ConverterService
+func NewService(config Config) (ConverterService, error) {
+	if config.CCKey == "" {
+		return ConverterService{}, errors.New("cckey cannot be empty")
 	}
 	return ConverterService{
-		APIKey: APIKey,
+		APIKey: config.CCKey,
+		URL:    config.URL,
 	}, nil
 }
 
-func (cs *ConverterService) Compare(have, want string) (*currconv.ConvertHistorical, error) {
-	//  line 25 - 30 would all happen in NewService
-	api := currconv.NewAPI(currconv.Config{
-		// Change base URL to be an env. variable
-		BaseURL: "https://free.currconv.com",
-		Version: "v7",
-		APIKey:  cs.APIKey,
-	})
-	conver, err := api.ConvertHistorical(currconv.ConvertHistoricalRequest{
-		Q:    []string{have + "_" + want},
-		Date: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-7, 0, 0, 0, 0, time.UTC),
-		// Create a data and subtract a date from it.
-		EndDate: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC),
-	})
+type ConvertHistorical struct {
+	Query struct {
+		Count int `json:"count"`
+	} `json:"query"`
+	Date    string `json:"date"`
+	EndDate string `json:"endDate,omitempty"`
+	Results map[string]struct {
+		ID  string             `json:"id"`
+		To  string             `json:"to"`
+		Fr  string             `json:"fr"`
+		Val map[string]float32 `json:"val"`
+	} `json:"results"`
+}
+
+// TODO: Compare needs to be entirely re-written to use the standard library.
+func (cs *ConverterService) Compare(have, want string) (*ConvertHistorical, error) {
+	var conver = ConvertHistorical{}
+	v := url.Values{}
+	v.Add("q", have+"_"+want)
+	pastdate := time.Now().AddDate(0, 0, -7)
+	date := fmt.Sprintf("%d-%d-%d", pastdate.Year(), pastdate.Month(), pastdate.Day())
+	endDate := fmt.Sprintf("%d-%d-%d", time.Now().Year(), time.Now().Month(), time.Now().Day())
+	v.Add("date", date)
+	v.Add("endDate", endDate)
+	v.Add("apiKey", cs.APIKey)
+	url := cs.URL + v.Encode()
+	resp, err := http.Get(url)
 	if err != nil {
-		return nil, fmt.Errorf("render: %w", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
```

Comment:
I like to use the http constants here for clarity, so would make it http.StatusOk